cli/daemon/dash/dash.go

```diff
@@ -152,25 +192,61 @@ func (h *handler) Handle(ctx context.Context, reply jsonrpc2.Replier, r jsonrpc2
 		}
 		return h.apiCall(ctx, reply, &params)
 
-	case "source-context":
+	case "editors/list":
+		var resp struct {
+			Editors []string `json:"editors"`
+		}
+
+		found, err := editors.Resolve(ctx)
+		if err != nil {
+			log.Err(err).Msg("dash: could not list editors")
+			return reply(ctx, nil, err)
+		}
+
+		for _, e := range found {
+			resp.Editors = append(resp.Editors, string(e.Editor))
+		}
+		return reply(ctx, resp, nil)
+
+	case "editors/open":
 		var params struct {
-			AppID string
-			File  string
-			Line  int
+			AppID     string             `json:"app_id"`
+			Editor    editors.EditorName `json:"editor"`
+			File      string             `json:"file"`
+			StartLine int                `json:"start_line,omitempty"`
+			StartCol  int                `json:"start_col,omitempty"`
+			EndLine   int                `json:"end_line,omitempty"`
+			EndCol    int                `json:"end_col,omitempty"`
 		}
 		if err := unmarshal(&params); err != nil {
+			log.Warn().Err(err).Msg("dash: could not parse open command")
 			return reply(ctx, nil, err)
 		}
-		f, err := os.Open(params.File)
+
+		editor, err := editors.Find(ctx, params.Editor)
 		if err != nil {
+			log.Err(err).Str("editor", string(params.Editor)).Msg("dash: could not find editor")
 			return reply(ctx, nil, err)
 		}
-		defer fns.CloseIgnore(f)
-		lines, start, err := sourceContext(f, params.Line, 5)
+
+		app, err := h.apps.FindLatestByPlatformOrLocalID(params.AppID)
 		if err != nil {
+			if errors.Is(err, apps.ErrNotFound) {
+				return reply(ctx, nil, fmt.Errorf("app not found, try running encore run"))
+			}
+			log.Err(err).Str("app_id", params.AppID).Msg("dash: could not find app")
 			return reply(ctx, nil, err)
 		}
-		return reply(ctx, sourceContextResponse{Lines: lines, Start: start}, nil)
+
+		params.File = filepath.Join(app.Root(), params.File)
```

Comment:
We should check that the input file is local (filepath.IsLocal) before this to ensure only files within the app are opened.